google-cloud-storage/src/main/java/com/google/cloud/storage/GapicUnbufferedReadableByteChannel.java

```diff
@@ -92,7 +87,12 @@ public long read(ByteBuffer[] dsts) throws IOException {
   @Override
   public long read(ByteBuffer[] dsts, int offset, int length) throws IOException {
     if (complete && open) {
-      close();
+      if (blobOffset != totalSize) {
```

Comment:
When would this happen?

Reply:
If there were some sort of corruption, or stream truncation that didn't result in a premature eof error.

Seems like a fairly simple check we can do regardless of whether crc32c validation is enabled or not.

Reply:
I'm assuming we would retry this failure, because it throws an IOException?

Reply:
Nah it's fine, as long as we retry it then it's okay.

Reply:
412 Isn't retriable right now. So, I'm going to remove this for now and we can add it back in when we come back and do retries.